### PR TITLE
fix: Hacky fix for running the experimental scheduler when spawning task

### PR DESCRIPTION
**Pull request by vrongmeal**
Until I figure out the best solution (which hopefully is close).

Pushing just to benchmark and test stuff successfully.

To run the tokio console debugger:

1. Download the CLI:
   ```
   cargo --locked tokio-console
   ```
2. Build/Run local with new just commands (along with `--debug-tokio` flag):
   ```
   just run-debug-tokio local --debug-tokio
   ```

**NOTE:** This builds in a new `target` directory: `unstable-tokio-target` so as to not mess up with existing target directory (which can cause issues with LSP etc.).

### CODE REVIEW

File: crates/glaredb/src/commands.rs [hunk 1]
```diff
@@ -55,6 +55,11 @@ trait RunCommand {
 
 impl RunCommand for LocalArgs {
     fn run(self) -> Result<()> {
+        if self.debug_tokio {
+            eprintln!("> Starting tokio debugger");
+            console_subscriber::init();
+        }
```
Comment (tychoish) [inline crates/glaredb/src/commands.rs:61]: is there an easy way to ifdef this out in release mode?

File: crates/sqlexec/src/distexec/scheduler.rs [hunk 2]
```diff
@@ -25,8 +25,28 @@ impl Scheduler {
...
+        // Due to our implementation, the runtime is blocked somehow and unable
+        // to spawn new tasks.
+        //
+        // HACK: Introducing a new task here doesn't block the runtime from
+        // spawning new tasks.
+        let (dummy_task_sender, dummy_task_receiver) = async_channel::unbounded();
+        tokio::spawn(async move {
+            while let Ok(task) = dummy_task_receiver.recv().await {
+                actual_task_sender.send(task).await.unwrap();
+            }
```
Comment (tychoish) [inline crates/sqlexec/src/distexec/scheduler.rs:39]: how fast would we expect this to thrash/spawn?

File: justfile [hunk 3]
```diff
@@ -29,6 +29,14 @@ run *args: protoc
...
   cargo build --bin glaredb {{args}}
 
+# Build glaredb with unstable_tokio flag.
+#
+# Need to run the local command with `--debug-tokio` argument.
+build-debug-tokio *args:
+  RUSTFLAGS="--cfg tokio_unstable" CARGO_TARGET_DIR=tokio-unstable-target just build {{args}}
+
+run-debug-tokio *args:
+  RUSTFLAGS="--cfg tokio_unstable" CARGO_TARGET_DIR=tokio-unstable-target just run {{args}}
```
Comment (tychoish) [inline justfile:39]: I might be _slightly_ partial to using `--target-dir` over `CARGO_TARGET_DIR` but it's probably not material

**Review by tychoish — COMMENTED** [review]
is it possible to require/switch to the new scheduler via the command line.

also we should add a task to add a new matrix dimension to the SLT tests that would run with the new scheduler. (x pg x flight once this land and after this lands.)

File: crates/glaredb/src/commands.rs (hunk 1 above)
Comment (vrongmeal) [reply crates/glaredb/src/commands.rs:61]: I'll do that!

File: crates/sqlexec/src/distexec/scheduler.rs (hunk 2 above)
Comment (vrongmeal) [reply crates/sqlexec/src/distexec/scheduler.rs:39]: Immediately!

File: justfile (hunk 3 above)
Comment (vrongmeal) [reply justfile:39]: Yeah, just wanted something quick.

**Comment by vrongmeal** [comment]
> is it possible to require/switch to the new scheduler via the command line.

No, unfortunately have to set it for the session.

File: crates/glaredb/src/commands.rs (hunk 1 above)
Comment (vrongmeal) [reply crates/glaredb/src/commands.rs:61]: Updated!

**Review by tychoish — APPROVED** [review]

**Pull request merged**